packages/proxy/src/proxy.ts

```diff
@@ -187,11 +198,29 @@ export async function proxyV1({
   }
 
   // Caching is enabled by default, but let the user disable it
-  const useCacheMode = parseEnumHeader(
+  // XXX is there a clean way to specify "auto" caching behavior using only cache-control directives?
+  const legacyCacheMode = parseEnumHeader(
     CACHE_HEADER,
     CACHE_MODES,
     proxyHeaders[CACHE_HEADER],
   );
+
+  let useCacheMode = legacyCacheMode;
+  const cacheControlDirectives = cacheControlParse(
+    proxyHeaders[CACHE_CONTROL_HEADER] || "",
+  );
+  // XXX does it make sense to have separate cache control for read and write?
+  if (
+    cacheControlDirectives?.["no-cache"] ||
+    cacheControlDirectives?.["no-store"]
+  ) {
+    useCacheMode = "never";
+  }
+  const cacheTTL = Math.min(
+    Math.max(1, cacheControlDirectives?.["max-age"] ?? DEFAULT_CACHE_TTL),
+    DEFAULT_CACHE_TTL,
```

Comment:
do you mean MAX_CACHE_TTL?

Reply:
also, i wonder if we should have a separate, bt-specific, header that controls the ttl since from the MDN docs, it doesn't seem like max-age controls the ttl on the server, just how fresh of a response is expected by the client